src/views/DatasetManager/components/DatasetOverview/components/CategoryAccordion/components/CategoryDetailsDialog/CategoryDetailsDialog.js

```diff
@@ -0,0 +1,162 @@
+      clearApplicationErrorMessage();
+    }
+    setOpen(false);
+  }, [clearApplicationErrorMessage, isThereAlreadyError]);
+
+  const errorBanner = useMemo(() => {
+    return (
+      <ErrorBanner
+        error={applicationError}
+        labels={{
+          dismissButtonText: t('commoncomponents.banner.button.dismiss', 'Dismiss'),
+          tooLongErrorMessage: t(
+            'commoncomponents.banner.tooLongErrorMessage',
+            'Detailed error message is too long to be displayed. ' +
+              'To read it, please use the COPY button and paste it in your favorite text editor.'
+          ),
+          secondButtonText: t('commoncomponents.banner.button.copy.label', 'Copy'),
+          toggledButtonText: t('commoncomponents.banner.button.copy.copied', 'Copied'),
+        }}
+        clearErrors={clearApplicationErrorMessage}
+      />
+    );
+  }, [applicationError, clearApplicationErrorMessage, t]);
+
+  const categoryKpis = useMemo(() => {
+    return (
+      <Grid container sx={{ p: 5 }} spacing={'64px'} justifyContent="center">
+        {category.kpis.map((kpi) => {
+          return (
+            <Grid item key={`${kpi.id}-key`}>
+              <Card sx={{ width: '250px', backgroundColor: '#ffe26b1c', height: '100%' }}>
+                <CardContent>
+                  <Typography variant="h5">
+                    {t(TranslationUtils.getDatasetCategoryKpiNameTranslationKey(category.id, kpi.id), kpi.id)}
+                  </Typography>
+                  <Grid item sx={{ mt: 3 }}>
+                    <KPIValue
+                      kpi={kpis.find((kpiResult) => kpiResult.id === kpi.id)}
+                      valueTypographyProps={{ variant: 'body2', sx: { opacity: '70%' } }}
+                    />
+                  </Grid>
+                </CardContent>
+              </Card>
+            </Grid>
+          );
+        })}
+      </Grid>
+    );
+  }, [category.id, category.kpis, kpis, t]);
+
+  const query = useMemo(
+    () => getQuery(category?.previewTable?.queryId)?.query,
+    [category?.previewTable?.queryId, getQuery]
+  );
+
+  const tableEdit = useMemo(() => {
+    if (open === false || query == null) return null;
+    const context = {
+      isDarkTheme,
+      editMode: false,
+      targetDatasetId: datasetId,
+    };
+
+    const parameterData = {
+      id: category?.id,
+      options: {
+        columns: category?.previewTable?.columns,
+        dynamicValues: {
+          query,
+          resultKey: 'fields',
+        },
+      },
+    };
+
+    const parameterValue = { status: TABLE_DATA_STATUS.EMPTY };
+
+    return (
+      <GenericTable
+        parameterData={parameterData}
+        context={context}
+        parameterValue={parameterValue}
+        setParameterValue={() => {}}
+        resetParameterValue={() => {}}
+      />
+    );
+  }, [query, open, isDarkTheme, datasetId, category?.id, category?.previewTable?.columns]);
+
+  return query == null ? null : (
```

Comment:
Maybe React syntax `query && ...` is slightly more readable ?

Reply:
In this case I need a ternary operator